src/Aspire.Hosting/Dashboard/proto/Partials.cs

```diff
@@ -1,13 +1,24 @@
 // Licensed to the .NET Foundation under one or more agreements.
 // The .NET Foundation licenses this file to you under the MIT license.
 
+using System.Collections.Immutable;
+using Aspire.Dashboard.Model;
 using Aspire.Hosting.Dashboard;
 using Google.Protobuf.WellKnownTypes;
 
 namespace Aspire.ResourceService.Proto.V1;
 
 partial class Resource
 {
+    /// <summary>
+    /// Names of resource properties known to contain potentially sensitive values that should be masked in the UI by default.
+    /// </summary>
+    private static readonly ImmutableHashSet<string> s_knownSensitiveResourcePropertyNames =
+    [
+        KnownProperties.Container.Args,
+        KnownProperties.Executable.Args,
+    ];
```

Comment:
These are the hard-coded properties that the app host considers to be sensitive for now. We can easily add others if needed.